packages/autocomplete-plugin-algolia-insights/src/createAlgoliaInsightsPlugin.ts

```diff
@@ -240,24 +240,32 @@ export function createAlgoliaInsightsPlugin(
 
 function getOptions(options: CreateAlgoliaInsightsPluginParams) {
   return {
-    onItemsChange({ insights, insightsEvents }: OnItemsChangeParams) {
+    onItemsChange({ insights, insightsEvents, state }: OnItemsChangeParams) {
       insights.viewedObjectIDs(
         ...insightsEvents.map((event) => ({
           ...event,
           algoliaSource: [
             ...(event.algoliaSource || []),
             'autocomplete-internal',
+            ...((state.context.algoliaInsightsPlugin as Record<string, unknown>)
+              ?.__automaticInsights
+              ? ['autocomplete-automatic']
+              : []),
           ],
```

Comment:
Since this is the same code here and in `onSelect`, maybe we can refactor it with a `getAlgoliaSource` function?

Reply:
Good idea, I'll do this.

Reply:
Done in https://github.com/algolia/autocomplete/pull/1213/commits/9a5377fa2ab978d2c77495bf11114ef522dcc4f0.